src/data.rs

```diff
@@ -0,0 +1,140 @@
+/// # Binary Field Encoding (BFE) Specification Data
+///
+/// This represents all the constant values used for BFE, defined in the [specification](https://github.com/ssb-ngi-pointer/ssb-bfe-spec/blob/master/bfe.json).
+///
+/// _Note_: not all of these type-format variants are currently supported in ssb-bfe-rs.
+use phf::{phf_ordered_map, OrderedMap};
+
+/// A tuple struct defining all possible BFE specification data points.
+pub struct BfeSpecData<'a>(
+    /// type-format code
+    pub &'a [u8],
+    /// data length
+    pub Option<usize>,
+    /// key length
```

Comment:
I was wondering what's the difference between `data length` and `key length` and I think this is stray code. The spec has https://github.com/ssb-ngi-pointer/ssb-bfe-spec/blob/a4e38c30b67d7110115c7fc8440b71448f87d8d2/bfe.json#L35 but it's not used anywhere, and I think the semantics are the same as `data`.

We need to update the spec to not have it.

And we need to update ssb-bfe js to not do this: https://github.com/ssb-ngi-pointer/ssb-bfe/blob/7ea8dfe38749d496d528ec0eac30e6e670e133fb/util.js#L17

So this PR could also get rid of this